### ISSUE TITLE
Evaluate RR pt2

### ISSUE THREAD

**Issue by koutsd** [ubicomplab/rPPG-Toolbox#246]
I was trying to reopen this issue but I could not so i opened a new one:
             
 Hi again,

Excuse my potential ignorance since I'm not very familiar with signal processing and PPG in general.

I have seen papers about extracting respiration rate from PPG signals. Do you think that those methods would be valid for the rPPG signal produced by the neural network models?

From what I understand from the trainer implementations, the NN models are trained using the PPG ground truth data as labels without applying any filter to remove the respiratory signal. So, shouldn't the predicted rPPG signal contain respiratory information?

_Originally posted by @koutsd in https://github.com/ubicomplab/rPPG-Toolbox/issues/241#issuecomment-1961294388_

**Comment by girishvn** [comment]
Hi @koutsd, 
Yes, respiratory sinus arrhythmia are the modulation of the human pulse as a result of respiration.
Though this phenomena exists, its effect is variable and depends on a number of factors (namely age and health). I would guess that the RSA signal is usable from contact ppg (like a pulse ox), but that the SNR is too low for it to be useful in a remote rppg scenario. The ground truth PPG signals should exhibit an RSA signal, but I cannot speak to its magnitude or quality. I would refer to https://github.com/xliucs/MTTS-CAN?tab=readme-ov-file to learn more.

The NN architecture apply a point-to-point color mapping between face color and PPG amplitude, so theoretically some information regarding RSA should also be encode. Practically, in my experience / observation no gross or well defined RSA signal presents its self (likely an SNR issue of the rPPG signal). Historically, people have extracted RR from video by observing body motion (the motion of the head / abdomen during inhalation and exhalation). And this has had moderate success. MTTSCAN claims to make use of these RSA's to improve both, which is possible, but the lack of improvement of multitask respiration over the single task respiration leads me to think the vast majority of the respiration signal is derived from body motion. 

For next steps, I would inspect the ground truth PPG signals for these datasets. If you are able to observe a .15 - 0.3 Hz modulation (potential respiration), then it is possible that a similar characteristic may be present its self in the NN outputs. 

Hope this helps!

**Comment by girishvn** [comment]
Hi, 

Its possible that you can use a supervised method for this, though I am unsure of the efficacy (it probably depends on the quality/lighting/etc. of the input data, and would guess that it would run into a number of issues similar to NN methods where SNR of the respiration signal may be a problem. 

With such a narrow freq range for possible RR its difficult to say if those results are actually meaningful.
The best way to determine the efficacy is to plot the predicted respiration signal against some ground truth.

**Comment by girishvn** [comment]
Closing this issue, feel free to re-open if need be.
